tasks/client.yaml

```diff
@@ -1,25 +1,154 @@
 ---
-- name: Operating system dependencies
-  apt: name={{ item }} state=present
+
+# dependencies
+
+- name: Dependencies are installed with Aptitude
+  apt: 
+    name: "{{ item }}"
+    state: present
+  with_items: "{{ _letsencrypt_apt_packages }}"
+  when: _letsencrypt_apt_packages is defined
+
+- name: Dependencies are installed with RPM
+  apt: 
+    name: "{{ item }}"
+    state: present
+  with_items: "{{ _letsencrypt_rpm_packages }}"
+  when: _letsencrypt_rpm_packages is defined
+
+
+# user and group configuration
+
+- name: letsencrypt_group exits
+  group:
+    name: "{{ letsencrypt_group }}"
+    system: yes
+    state: present
+
+- name: letsencrypt_user exists
+  user:
+    name: "{{ letsencrypt_user }}"
+    group: "{{ letsencrypt_group }}"
+    system: yes
+    home: "{{ letsencrypt_home_dir }}"
+    move_home: no
+    createhome: yes
+    append: yes
+
+- name: letsencrypt_home_dir is writable for letsencrypt_user
+  file:
+    path: "{{ letsencrypt_home_dir }}"
+    state: directory
+    mode: 0770
+
+
+# file system access
+
+- name: letsencrypt user is able to write to output directories
+  file:
+    path: "{{ item }}"
+    recurse: yes
+    owner: "{{ letsencrypt_user }}"
   with_items:
-    - build-essential
-    - libssl-dev
-    - libffi-dev
-    - python-dev
-    - git
-    - python-pip
-    - python-virtualenv
-    - dialog
-    - libaugeas0
-    - ca-certificates
-- name: Python cryptography module
-  pip: name=cryptography
-- name: Letsencrypt Python client
+    - /var/lib/letsencrypt
+    - /var/log/letsencrypt
+    - /etc/letsencrypt
+
+- name: letsencrypt account directory exists and is writable
+  file:
+    path: "{{ letsencrypt_account_file | dirname }}"
+    owner: root
+    group: "{{ letsencrypt_group }}"
+    mode: 0770
+    state: directory
+
+- name: letsencrypt_webroot_path exists and is writable by letsencrypt_user
+  file:
+    path: "{{ letsencrypt_webroot_path }}"
+    state: directory
+    owner: "{{ letsencrypt_user }}"
+    group: "{{ letsencrypt_group }}"
+    mode: 0775
+    recurse: yes
+
+
+# python webserver systemd wrapper
+
+- name: webserver systemd service is installed
```

Comment:
When `ansible_service_mgr == "upstart"` an upstart job will be deployed instead of a systemd service. Tested on 14.04.